src/machinery/eventListener.ts

```diff
@@ -23,18 +24,15 @@ let stateHistory: NavigationState[] = [START_STATE];
 
 let index = 0;
 
-/** setTimeout hack needed to wire up navigation after DOM is loaded, there's probably a better way */
-navigationStore.subscribe(() => {
```

Comment:
This was error prone, now explicitly calls navigationReload when components have changed.